sync/src/main/java/tech/pegasys/artemis/sync/PeerSync.java

```diff
@@ -63,6 +67,10 @@ public PeerSync(
     final UnsignedLong latestFinalizedSlot = compute_start_slot_at_epoch(finalizedEpoch);
     final UnsignedLong firstNonFinalSlot = latestFinalizedSlot.plus(UnsignedLong.ONE);
 
+    this.startingSlot = firstNonFinalSlot;
+    this.currentSlot = startingSlot;
+    this.highestSlot = peer.getStatus().getHeadSlot();
```

Comment:
This needs to be in a synchronised block.  Both the read and write sides have to be synchronised for it to have any effect.